### PR TITLE
fix: space bug when in pip

### LINKED ISSUE #220: Media Viewer: Activating picture-in-picture mode breaks use of spaces when typing messages

### PR DESCRIPTION
**Pull request by iib0011**
fixes #220

### CODE REVIEW

File: src/hooks/usePictureInPicture.ts [hunk 1]
```diff
@@ -28,8 +29,8 @@ export default function usePictureInPicture(
     // @ts-ignore
     video.autoPictureInPicture = true;
     setIsSupported(true);
-    video.addEventListener('enterpictureinpicture', onEnter);
-    video.addEventListener('leavepictureinpicture', onLeave);
+    video.addEventListener('enterpictureinpicture', () => { onEnter(); setIsInPictureInPicture(true); });
+    video.addEventListener('leavepictureinpicture', () => { onLeave(); setIsInPictureInPicture(false); });
```
Comment (Ajaxy) [inline src/hooks/usePictureInPicture.ts:33]: Introducing an arrow function here will break `removeEventListener`.

**Review by Ajaxy — CHANGES_REQUESTED** [review]

File: src/hooks/usePictureInPicture.ts (hunk 1 above)
Comment (iib0011) [reply src/hooks/usePictureInPicture.ts]: fixed

File: src/hooks/usePictureInPicture.ts [hunk 2]
```diff
@@ -16,9 +16,16 @@ export default function usePictureInPicture(
   onLeave: CallbackType,
 ): ReturnType {
   const [isSupported, setIsSupported] = useState(false);
+  const [isInPictureInPicture, setIsInPictureInPicture] = useState(false);
 
   useLayoutEffect(() => {
     // PIP is not supported in PWA on iOS, despite being detected
```
Comment (Ajaxy) [inline src/hooks/usePictureInPicture.ts:22]: This comment now looks non-relevant.

**Review by Ajaxy — CHANGES_REQUESTED** [review]

File: src/hooks/usePictureInPicture.ts [hunk 3]
```diff
@@ -28,11 +35,11 @@ export default function usePictureInPicture(
     // @ts-ignore
     video.autoPictureInPicture = true;
     setIsSupported(true);
-    video.addEventListener('enterpictureinpicture', onEnter);
-    video.addEventListener('leavepictureinpicture', onLeave);
+    video.addEventListener('enterpictureinpicture', onEnterInternal);
```
[the remark this answers is not in the document]
Comment (iib0011) [reply src/hooks/usePictureInPicture.ts:38]: done

File: src/hooks/usePictureInPicture.ts [hunk 4]
```diff
@@ -28,11 +29,17 @@ export default function usePictureInPicture(
     // @ts-ignore
     video.autoPictureInPicture = true;
     setIsSupported(true);
-    video.addEventListener('enterpictureinpicture', onEnter);
-    video.addEventListener('leavepictureinpicture', onLeave);
+    const onEnterInternal = () => {
+      onEnter(); setIsInPictureInPicture(true);
```
Comment (Ajaxy) [inline src/hooks/usePictureInPicture.ts:33]:
```suggestion
      onEnter();
      setIsInPictureInPicture(true);
```

File: src/hooks/usePictureInPicture.ts [hunk 5]
```diff
@@ -28,11 +29,17 @@ export default function usePictureInPicture(
...
+    };
+    const onLeaveInternal = () => {
+      onLeave(); setIsInPictureInPicture(false);
```
Comment (Ajaxy) [inline src/hooks/usePictureInPicture.ts:36]:
```suggestion
      onLeave();
      setIsInPictureInPicture(false);
```

**Review by Ajaxy — APPROVED** [review]

**Comment by Ajaxy** [comment]
Merged, thank you!

**Pull request merged**